.gitignore

```diff
@@ -11,3 +11,7 @@ node_modules/
 static/dist
 settings.json
 launch.json
+.idea
+package-lock.json
```

Comment:
We generally want to have package-lock.jsons. What's the purpose of this and the lines below?

Reply:
There's no package-lock.json checked in so I assumed it was generated by NPM at runtime.

The .js and .js.map lines are for all the compiled JS files that Vue/Typescript poop out in the source directory

Reply:
Ah, we have a slightly weird repo structure. The package-locks are checked in to `/client/` and `/client-toolchain/` (but we do still want to check them in, they're the source of truth for our dependency versioning).

The repo actually has at least one real `.js` file, so we don't want a blanket ban on js files. We should already be ignoring all the compiler-generated .js files though (they go in `/client_toolchain/dev_server/dist` and `/static/dist`). Are you seeing them pop up elsewhere?

Reply:
Ok, this might be IntelliJ putting stuff in weird places then.